src/core/i18n/index.tsx

```diff
@@ -9,7 +9,7 @@ export * from './utils';
 
 i18n.use(initReactI18next).init({
   resources,
-  lng: getLanguage() ?? locale, // TODO: if you are not supporting multiple languages or languages with multiple directions you can set the default value to `en`
+  lng: getLanguage() ?? getLocales()[0].languageTag, // TODO: if you are not supporting multiple languages or languages with multiple directions you can set the default value to `en`
```

Comment:
I wonder how this behaves when the user changes the OS language (and `getLanguage()` still returns null) after the app was open.

[Expo suggest to listen to app state changes](https://docs.expo.dev/versions/latest/sdk/localization/#behavior) and rerun the `getLanguage` function. In our case we would use `changeLanguage` function from `i18next` and set the return value of this same expression (`getLanguage() ?? getLocales()[0].languageTag`).

Of course this is something we can test deeply and, if it applies, improve later. It's is completely out of the scope of this PR and task. Just a thought.

Reply:
Good question.
I like the approach of listen app state changes, if the app comes from background to active then getLocales should be called manually.

Reply:
I've created a task to our board: https://www.notion.so/rootstrap/Change-Language-based-on-OS-language-12959347eba680339165db5f963b9535?pvs=4